modin/polars/series.py

```
@@ -0,0 +1,2151 @@
+
+    def log10(self) -> "Series":
+        """
+        Calculate the base 10 logarithm values.
+
+        Returns:
+            Base 10 logarithm values Series.
+        """
+        return self.log(10)
+
+    def log1p(self) -> "Series":
+        """
+        Calculate the natural logarithm of 1 plus the values.
+
+        Returns:
+            Natural logarithm of 1 plus the values Series.
+        """
+        raise NotImplementedError("not yet")
+
+    def replace(
+        self,
+        mapping: dict[Any, Any],
+        *,
+        default: Any = None,
+        return_dtype=None,
+    ) -> "Series":
+        """
+        Map values to other values.
+
+        Args:
+            mapping: Mapping.
+
+        Returns:
+            Mapped Series.
+        """
+        return self.__constructor__(
+            self.to_pandas().apply(lambda x: mapping.get(x, default))
+        )
+
+    def pct_change(self, n: int = 1) -> "Series":
+        """
+        Calculate the percentage change.
+
+        Args:
+            n: Number of periods to shift.
+
+        Returns:
+            Percentage change Series.
+        """
+        return self.__constructor__(self.to_pandas().pct_change(n))
+
+    def peak_max(self) -> "Series":
+        """
+        Get the peak maximum values.
+
+        Returns:
+            Peak maximum values Series.
+        """
+        return self.__eq__(self.max())
+
+    def peak_min(self) -> "Series":
+        """
+        Get the peak minimum values.
+
+        Returns:
+            Peak minimum values Series.
+        """
+        return self.__eq__(self.min())
+
+    def rank(
+        self,
+        method: str = "average",
+        *,
+        descending: bool = False,
+        seed: int | None = None,
+    ) -> "Series":
+        """
+        Calculate the rank.
+
+        Args:
+            method: Rank method.
+
+        Returns:
+            Rank Series.
+        """
+        return self.__constructor__(
```

Should we raise `NotImplementedError` for unsupported `method`?